tests/PHPStan/Analyser/data/array-combine.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace ArrayCombine;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+	/** @phpstan-return 'foo' */
+	public function __toString(): string
+	{
+		return 'foo';
+	}
+}
+
+class Bar
+{
+	public function __toString(): string
+	{
+		return 'bar';
+	}
+}
+
+class Baz {}
+
+function withBoolKey(): void
+{
+	$a = [true, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+
+	$c = [false, 'red', 'yellow'];
+	$d = ['avocado', 'apple', 'banana'];
+
+	assertType("array{: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($c, $d));
+}
+
+function withFloatKey(): void
+{
+	$a = [1.5, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1.5: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+}
+
+function withIntegerKey(): void
+{
+	$a = [1, 2, 3];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', 2: 'apple', 3: 'banana'}", array_combine($a, $b));
+}
+
+function withNumericStringKey(): void
+{
+	$a = ["1", "2", "3"];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', 2: 'apple', 3: 'banana'}", array_combine($a, $b));
+}
+
+function withObjectKey() : void
+{
+	$a = [new Foo, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{foo: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+	assertType("non-empty-array<string, 'apple'|'avocado'|'banana'>", array_combine([new Bar, 'red', 'yellow'], $b));
```

Comment:
I think there should be another test, for having such an object within the $values array

https://3v4l.org/gWPIs

Reply:
Why? There is nothing done special with values array. It can be any type.

Reply:
To make sure phpstan represents the objects as steing type in values.

I feel we should maybe even have a rule which errors when passing a object here (if we don't have one already)

Reply:
I don't understand. Objects in values are not casted to string. Like in the 3v4l link you shared.

Reply:
Ouch,  you are right. Misread the output